kuma/core/models.py

```diff
@@ -34,6 +37,14 @@ def update(self, **kw):
                                           created=False)
 
 
+class IPBan(models.Model):
+    ip = models.CharField(max_length=40, editable=False, db_index=True,
+                          blank=True, null=True, unique=True)
+    created = models.DateTimeField(default=datetime.now, db_index=True)
```

Comment:
Re: the caching comment, I've been doing something like:

```python
@classmethod
def get(cls, some_natural_key):
    obj = cache.get(cls._cache_key(some_natural_key))
    if obj == EMPTY_SENTINEL:  # e.g. '-'
        raise cls.DoesNotExist  # or return None
    # ... etc
```

IP address, and `deleted__isnull` make a pretty natural key here.